tests/components/hatchrest/__init__.py

```diff
@@ -0,0 +1,32 @@
+"""Tests for the hatchrest integration."""
+from bleak.backends.device import BLEDevice
+from bleak.backends.scanner import AdvertisementData
+
+from homeassistant.components.bluetooth import BluetoothServiceInfoBleak
+from homeassistant.components.hatchrest.const import DOMAIN
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.helpers.device_registry import format_mac
+
+from tests.common import MockConfigEntry
+
+VALID_HATCHREST = BluetoothServiceInfoBleak.from_advertisement(
+    BLEDevice("00:00:00:00:00:00", "Hatch Baby Rest"),
+    AdvertisementData(
+        local_name="Hatch Baby Rest",
+        address="00:00:00:00:00:00",
```

Comment:
I'm not sure about all the tests, but some of them (e.g., https://github.com/home-assistant/core/actions/runs/3696984389/jobs/6261629900) seem to hint towards that this line is the problem:
```
 tests/components/hatchrest/__init__.py:14: in <module>
    AdvertisementData(
E   TypeError: AdvertisementData.__new__() got an unexpected keyword argument 'address'
```
At least the current bleak master branch does not have this argument either: https://github.com/hbldh/bleak/blob/develop/bleak/backends/scanner.py#L22

Maybe fixing that would fix the CI?

Reply:
Looks like the APIs have changed a bit since I wrote it. I'm not noticing tests for anything else using Bluetooth yet, so I'm actually going to remove these tests since they seem flaky and comments on the namedtuples used say the APIs are unstable at the moment.

Reply:
Never mind. I see how the switchbot tests work now and updated mine.